graphql/resolvers.go

```diff
@@ -1392,21 +1392,37 @@ func (r *queryResolver) CommitQueue(ctx context.Context, id string) (*restModel.
 	if project.CommitQueue.Message != "" {
 		commitQueue.Message = &project.CommitQueue.Message
 	}
-	patchIds := []string{}
-	for _, item := range commitQueue.Queue {
-		issue := *item.Issue
-		patchIds = append(patchIds, issue)
-	}
-	patches, err := r.sc.FindPatchesByIds(patchIds)
-	if err != nil {
-		return nil, InternalServerError.Send(ctx, fmt.Sprintf("error finding patch: %s", err.Error()))
-	}
-	for i := range commitQueue.Queue {
-		for j := range patches {
-			if *commitQueue.Queue[i].Issue == *patches[j].Id {
-				commitQueue.Queue[i].Patch = &patches[j]
+
+	if project.CommitQueue.PatchType == commitqueue.CLIPatchType {
+		patchIds := []string{}
+		for _, item := range commitQueue.Queue {
+			issue := *item.Issue
+			patchIds = append(patchIds, issue)
+		}
+		patches, err := r.sc.FindPatchesByIds(patchIds)
+		if err != nil {
+			return nil, InternalServerError.Send(ctx, fmt.Sprintf("error finding patches: %s", err.Error()))
+		}
+		for i := range commitQueue.Queue {
+			for j := range patches {
+				if *commitQueue.Queue[i].Issue == *patches[j].Id {
+					commitQueue.Queue[i].Patch = &patches[j]
+				}
 			}
 		}
+	} else {
+		if len(commitQueue.Queue) > 0 {
+			versionId := restModel.FromStringPtr(commitQueue.Queue[0].Version)
+			if versionId != "" {
+				p, err := r.sc.FindPatchById(versionId)
+				if err != nil {
+					return nil, InternalServerError.Send(ctx, fmt.Sprintf("error finding patch: %s", err.Error()))
```

Comment:
This might be better as a NotFound since the most common error returned by FindPatchById will probably be that